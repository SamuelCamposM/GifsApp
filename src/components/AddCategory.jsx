import { useState } from "react";
import PropTypes from "prop-types";
export const AddCategory = ({ onNewCategory }) => {
  const [newCategory, setNewCategory] = useState("");
  const onInputChange = (e) => setNewCategory(e.target.value);
  const onSubmit = (e) => {
    e.preventDefault();
    if (newCategory.trim().length <= 1) return;
    // setCategories((prev) => [...prev, newCategory]);
    onNewCategory(newCategory.trim());
    setNewCategory("");
  };
  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Search gifs"
        value={newCategory}
        onChange={onInputChange}
      />
    </form>
  );
};
AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
export default AddCategory;
