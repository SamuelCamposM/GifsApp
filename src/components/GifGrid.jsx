import GifItem from "./GifItem";
import useFetchGifs from "../hooks/useFetchGifs";
import PropTypes from "prop-types";
export const GifGrid = ({ category }) => {
  const { isLoading, dataGifs } = useFetchGifs(category);
  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Cargando...</h2>}
      <div className="card-grid">
        {dataGifs.map((gif) => (
          <GifItem key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
export default GifGrid;
