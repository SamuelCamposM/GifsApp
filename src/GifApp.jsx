import React, { useState } from "react";
import { AddCategory, GifGrid } from "./components";
const GifApp = () => {
  const [categories, setCategories] = useState(["Valorant"]);
  const onAddCategory = (value) => {
    const exist = categories.some((category) => category == value);
    if (exist) return;
    setCategories((prev) => [value, ...prev]);
  };
  return (
    <>
      <h1>Gif App</h1>
      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid category={category} key={category} />
      ))}
    </>
  );
};

export default GifApp;
