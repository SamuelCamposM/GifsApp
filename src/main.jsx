import React from "react";
import ReactDOM from "react-dom/client"; 
import GifApp from "./GifApp.jsx";
import "./styles.css"
ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <GifApp />
  </React.StrictMode>
);
// 3baLos8gphAdZ7v9Z6gUcrswba3dgkuE