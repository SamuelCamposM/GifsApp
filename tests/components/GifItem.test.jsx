import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("test on <GifItem.jsx />", () => {
  const url = `https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/a67ac30c-fe21-4e97-8dad-ffa8b2670167/dch868l-316e23fd-b0cc-4ccb-ac88-eb71c06b87c3.gif?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwic3ViIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsImF1ZCI6WyJ1cm46c2VydmljZTpmaWxlLmRvd25sb2FkIl0sIm9iaiI6W1t7InBhdGgiOiIvZi9hNjdhYzMwYy1mZTIxLTRlOTctOGRhZC1mZmE4YjI2NzAxNjcvZGNoODY4bC0zMTZlMjNmZC1iMGNjLTRjY2ItYWM4OC1lYjcxYzA2Yjg3YzMuZ2lmIn1dXX0.PHPWw8pPy0L9gxxM5j1qCTs6F5YcEOQcgDMqUxVGHGI`;
  const title = "Yang Xiao Long";

  test("should match with the snapshot", () => {
    const { container } = render(<GifItem url={url} title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("should show the image with the URL and Alt indicated", () => {
    render(<GifItem url={url} title={title} />);
    // screen.debug();
    expect(screen.getByRole("img").alt).toBe(title);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should show the title with the component", () => {
    render(<GifItem url={url} title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
