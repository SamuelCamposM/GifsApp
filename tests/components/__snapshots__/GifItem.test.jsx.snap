// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <GifItem.jsx /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Yang Xiao Long"
      src="https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/a67ac30c-fe21-4e97-8dad-ffa8b2670167/dch868l-316e23fd-b0cc-4ccb-ac88-eb71c06b87c3.gif?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwic3ViIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsImF1ZCI6WyJ1cm46c2VydmljZTpmaWxlLmRvd25sb2FkIl0sIm9iaiI6W1t7InBhdGgiOiIvZi9hNjdhYzMwYy1mZTIxLTRlOTctOGRhZC1mZmE4YjI2NzAxNjcvZGNoODY4bC0zMTZlMjNmZC1iMGNjLTRjY2ItYWM4OC1lYjcxYzA2Yjg3YzMuZ2lmIn1dXX0.PHPWw8pPy0L9gxxM5j1qCTs6F5YcEOQcgDMqUxVGHGI"
    />
    <p>
      Yang Xiao Long
    </p>
  </div>
</div>
`;
